test/catalog/constraint_statement_full_test.cpp

```diff
@@ -0,0 +1,395 @@
+#include "traffic_cop/traffic_cop.h"
+
+#include <memory>
+#include <pqxx/pqxx>  // NOLINT
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "common/settings.h"
+#include "gtest/gtest.h"
+#include "main/db_main.h"
+#include "network/connection_handle_factory.h"
+#include "network/terrier_server.h"
+#include "storage/garbage_collector.h"
+#include "test_util/manual_packet_util.h"
+#include "test_util/test_harness.h"
+#include "traffic_cop/traffic_cop_defs.h"
+#include "transaction/deferred_action_manager.h"
+#include "transaction/transaction_manager.h"
+
+namespace terrier::trafficcop {
+
+    class ConstraintStatementTest : public TerrierTest {
+    protected:
+        void SetUp() override {
+            std::unordered_map<settings::Param, settings::ParamInfo> param_map;
+            terrier::settings::SettingsManager::ConstructParamMap(param_map);
+
+            db_main_ = terrier::DBMain::Builder()
+                    .SetSettingsParameterMap(std::move(param_map))
+                    .SetUseSettingsManager(true)
+                    .SetUseGC(true)
+                    .SetUseCatalog(true)
+                    .SetUseGCThread(true)
+                    .SetUseTrafficCop(true)
+                    .SetUseStatsStorage(true)
+                    .SetUseLogging(true)
+                    .SetUseNetwork(true)
+                    .SetUseExecution(true)
```

Comment:
Code style not quite consistent with the terrier code base. Consider enabling `ClangFormat` in CLion and then reformat code.